src/components/HomeSection.jsx

```diff
@@ -12,11 +12,15 @@ const HomeSection = ({
   pic,
 }) => {
   const picSize = main ? '95%' : '85%';
+  const textSize = main
```

Comment:
This is doesn't really address the root of the problem, which is the font sizes of different header variants.
Have you looked at this?

https://mui.com/customization/typography/#responsive-font-sizes

`responsiveFontSizes(theme)` seems to be what we need. You might need to change the variants of various typography elements across the app after making the font sizes responsive.